packages/aws-cdk/lib/api/logs/register-cloudwatch-logs.ts

```diff
@@ -0,0 +1,90 @@
+import * as cxapi from '@aws-cdk/cx-api';
+import { Mode, SdkProvider, ISDK } from '../aws-auth';
+import { EvaluateCloudFormationTemplate, LazyListStackResources } from '../evaluate-cloudformation-template';
+
+// resource types that have associated CloudWatch Log Groups that should
+// _not_ be monitored
+const IGNORE_LOGS_RESOURCE_TYPES = ['AWS::EC2::FlowLog', 'AWS::CloudTrail::Trail', 'AWS::CodeBuild::Project'];
+
+// resource types that will create a cloudwatch log group
+// with a specific name if one is not provided
+const RESOURCE_TYPES_WITH_IMPLICIT_LOGS = ['AWS::Lambda::Function'];
+
+export async function registerCloudWatchLogGroups(
```

Comment:
I think `findCloudWatchLogGroups()` might be a better name for this function, now that we refactored it to not mutate the CloudWatch logs monitor.

Will also require changing the name of this file, and the test file.